zerver/actions/invites.py

```diff
@@ -362,6 +363,64 @@ def do_get_invites_controlled_by_user(user_profile: UserProfile) -> list[dict[st
     return invites
 
 
+def get_invite_controlled_by_user(prereg_user: PreregistrationUser, user_profile: UserProfile) -> dict[str,Any]:
+    """
+    Returns the Invite representing invitation that can be controlled by user_profile.
+    This isn't necessarily the same as all the invitations generated by the user, as administrators
+    can control also invitations that they did not themselves create.
+    """
+    invitee = prereg_user;
+    if(invitee.include_realm_default_subscriptions):
+        stream_ids = list(get_default_stream_ids_for_realm(user_profile.realm))
+    else:
+        stream_ids = list(invitee.streams.values_list('id', flat=True))
+
+    invite = dict(
+                email=invitee.email,
+                invited_by_user_id=invitee.referred_by.id,
+                invited=datetime_to_timestamp(invitee.invited_at),
+                expiry_date=get_invitation_expiry_date(invitee.confirmation.get()),
+                id=invitee.id,
+                invited_as=invitee.invited_as,
+                is_multiuse=False,
+                notify_referrer_on_join=invitee.notify_referrer_on_join,
+                stream_ids=stream_ids
+            )
+    return invite
```

Comment:
Where is this logic checking if the user controls the invite? The function name suggests that this function is doing that.